test/micro/org/openjdk/bench/jdk/incubator/vector/ByteBufferVectorAccess.java

```diff
@@ -0,0 +1,152 @@
+import org.openjdk.jmh.annotations.Param;
+import org.openjdk.jmh.annotations.Setup;
+import org.openjdk.jmh.annotations.State;
+import org.openjdk.jmh.annotations.Warmup;
+
+@BenchmarkMode(Mode.AverageTime)
+@Warmup(iterations = 5, time = 500, timeUnit = TimeUnit.MILLISECONDS)
+@Measurement(iterations = 10, time = 500, timeUnit = TimeUnit.MILLISECONDS)
+@State(org.openjdk.jmh.annotations.Scope.Benchmark)
+@OutputTimeUnit(TimeUnit.NANOSECONDS)
+@Fork(value = 1, jvmArgsAppend = {
+    "--add-modules=jdk.incubator.foreign,jdk.incubator.vector",
+    "-Dforeign.restricted=permit",
+    "--enable-native-access", "ALL-UNNAMED"})
+public class ByteBufferVectorAccess {
+  private static final VectorSpecies<Byte> SPECIES = VectorSpecies.ofLargestShape(byte.class);
+
+  @Param("1024")
+  private int size;
+
+  ByteBuffer directIn, directOut;
+  ByteBuffer heapIn, heapOut;
+
+  ByteBuffer directInRo, directOutRo;
+  ByteBuffer heapInRo, heapOutRo;
+
+  @Setup
+  public void setup() {
+    directIn = ByteBuffer.allocateDirect(size);
+    directOut = ByteBuffer.allocateDirect(size);
+
+    heapIn = ByteBuffer.wrap(new byte[size]);
+    heapOut = ByteBuffer.wrap(new byte[size]);
+
+    directInRo = directIn.asReadOnlyBuffer();
+    directOutRo = directOut.asReadOnlyBuffer();
+
+    heapInRo = heapIn.asReadOnlyBuffer();
+    heapOutRo = heapOut.asReadOnlyBuffer();
+  }
+
+  @Benchmark
+  public void directBuffers() {
+    copyMemory(directIn, directOut);
+  }
+
+  @Benchmark
+  public void heapBuffers() {
+    copyMemory(heapIn, heapOut);
+  }
+
+  @Benchmark
+  public void pollutedBuffers2() {
+    copyMemory(directIn, directOut);
+    copyMemory(heapIn, heapOut);
+  }
+
+  @Benchmark
+  public void pollutedBuffers3() {
+    copyMemory(directIn, directOut);
+    copyMemory(heapIn, heapOut);
+
+    copyMemory(directInRo, directOut);
+    copyMemory(heapInRo, heapOut);
+  }
+
+  @Benchmark
+  public void pollutedBuffers4() {
+    copyMemory(directIn, heapOut); // Pollute if unswitch on 2nd param
+    copyMemory(heapIn, heapOut);
+
+    copyMemory(heapIn, directIn); // Pollute if unswitch on 1st param
+    copyMemory(heapIn, directOut);
+  }
+
+
+  boolean readOnlyException;
+
+  @Benchmark
+  public void pollutedBuffers5() {
+    copyMemory(directIn, heapOut);
+    copyMemory(heapIn, heapOut);
+
+    copyMemory(heapIn, directIn);
+    copyMemory(heapIn, directOut);
+
+    if (readOnlyException) {
+      try {
+        copyMemory(heapIn, directOutRo);
+      } catch (Exception ignored) {}
+      readOnlyException = !readOnlyException;
+    }
+  }
+
+  @Benchmark
+  public void arrayCopy() {
+    byte[] in = heapIn.array();
+    byte[] out = heapOut.array();
+
+    for (int i=0; i < SPECIES.loopBound(in.length); i += SPECIES.vectorByteSize()) {
+      final var v = ByteVector.fromArray(SPECIES, in, i);
+      v.intoArray(out, i);
+    }
+  }
+
+  @CompilerControl(CompilerControl.Mode.DONT_INLINE)
+  protected void copyMemory(ByteBuffer in, ByteBuffer out) {
```

Comment:
We may want to create a wrapping method that is not inlined just for the polluted cases e.g. `copyIntoNotInlined` calling `copyMemory`?

Thereby we avoid the measurement of a method call in the non-polluted cases so as to compare more fairly with `arrayCopy`?